.github/workflows/deploy.yml

```
@@ -1,28 +1,83 @@
+env:
+  CACHE_NAME: marlowe-temp
+  ALLOWED_URIS: "https://github.com https://api.github.com"
+  TRUSTED_PUBLIC_KEYS: "cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= marlowe-temp.cachix.org-1:1gPjVFpu4QjaAT3tRurCioX+BC23V7mjvFwpP5bV0Ec= loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk="
+  SUBSTITUTERS: "https://cache.nixos.org/ https://cache.iog.io https://marlowe-temp.cachix.org https://cache.zw3rk.com/"
+  VAULT_ADDR: "https://vault.dapps.aws.iohkdev.io"
+  NOMAD_ADDR: "https://nomad.dapps.aws.iohkdev.io"
+  NOMAD_NAMESPACE: "marlowe"
+
 on:
   push:
     branches:
-      - main # staging
-      - production
-      - deploy-hernan
-      - deploy-pablo
-      - shlevy
-      - marlowe-pioneers
-      - sprint-*
+      - main
 jobs:
   tests:
     uses: ./.github/workflows/test.yml
-  deploy:
+
+  publish:
     needs: [tests]
     runs-on: ubuntu-latest
-    env:
-      GITHUB_TOKEN: ${{ secrets.PLUTUS_OPS_GITHUB_TOKEN }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Inject slug/short variables
-        uses: rlespinasse/github-slug-action@v3.x
+      - name: Install Nix
+        uses: cachix/install-nix-action@v20
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          install_url: https://releases.nixos.org/nix/nix-2.10.3/install
+          extra_nix_config: |
+            allowed-uris = ${{ env.ALLOWED_URIS }}
+            trusted-public-keys = ${{ env.TRUSTED_PUBLIC_KEYS }}
+            substituters = ${{ env.SUBSTITUTERS }}
+            experimental-features = nix-command flakes
+
+      - name: Install Skopeo
+        run: nix-env -iA skopeo -f '<nixpkgs>'
+
+      - name: Login with Skopeo
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | skopeo login ghcr.io -u $ --password-stdin
+
+      - name: Publish Runtime Images
+        run: |
+          nix build .#oci-images.x86_64-linux.latest.all.copyToRegistry
+          ./result/bin/copy-to-registry
+
+  deploy:
+    strategy:
+      matrix:
+        network: [ preview, preprod, mainnet ]
+    needs: [publish]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Nix
+        uses: cachix/install-nix-action@v20
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          install_url: https://releases.nixos.org/nix/nix-2.10.3/install
+          extra_nix_config: |
+            allowed-uris = ${{ env.ALLOWED_URIS }}
+            trusted-public-keys = ${{ env.TRUSTED_PUBLIC_KEYS }}
+            substituters = ${{ env.SUBSTITUTERS }}
+            experimental-features = nix-command flakes
+
+      - name: Setup Cachix
+        uses: cachix/cachix-action@v12
+        with:
+          name: ${{ env.CACHE_NAME }}
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+
+      - name: Build nomad chart
+        run: |
+          nix build github:input-output-hk/dapps-world#marlowe.marlowe-runtime-${{ matrix.network }}-latest
```

```suggestion
          nix build github:input-output-hk/dapps-world#marlowe.marlowe-runtime-${{ matrix.network }}-latest --override-input marlowe-cardano github:input-output-hk/marlowe-cardano/$GITHUB_SHA
```
Then the action will deploy off of the commit that triggered it.

---

Beutiful!